### DISCUSSION TITLE
Create page template and load specific component on it

### DISCUSSION THREAD

**Discussion by comsup** [flyntwp/flynt#460] [Q&A] [answered]
Hi there and thanks for your awsome work.
I'm looking for a way to create a page template, like the one we use normally on wordpress and load some specifics components on it, but I can't figure a way to create a template and show it on the wordpress backend. 
I tried to create a file name "template-home.php" by duplicating the "page.php" and after that I have create a specific template in the templates folder and loaded it on my "template-home.php" but the template is not showing in WP backend. 

Is there a way to do that and is there a specific place to register my custom page template?

Many thanks !

**Comment by aaronmeder** [comment, marked as answer]
Hi @comsup 

That shouldn't differ from the [standard procedure](https://developer.wordpress.org/themes/template-files-section/page-template-files/) with WP.

1. Create the Template file, e.g. `template-specialsite.php `in your Flynt theme folder
2. Add a comment with the "Template Name: Special Site Template" to the header of the file. Example:
```
<?php

/* Template Name: Special Site Template */

use Timber\Timber;
use Timber\Post;

$context = Timber::get_context();
$context['post'] = new Post();

Timber::render('templates/page.twig', $context);
```

3. Then proceed by setting up and loading a specific twig-template if you want, as you already described it.

This will make it show up in Wordpress Admin:
<img width="500" alt="Screenshot 2022-09-13 at 11 00 34" src="https://user-images.githubusercontent.com/5564163/189860252-8d63380b-5e2e-446b-ae7f-3073709f7d8f.png">


Does that help?

Greetings from Zurich 🇨🇭
Aaron

**Reply by comsup** [reply]
I don't know why but I didn't put the wp comment code :) Morning work.. thanks !!

**Reply by aaronmeder** [reply]
Yeah a comment looks really unimportant - actually funny it was established like this in WP 😉
Glad that helped!

**Reply by comsup** [reply]
May be you can also help me with that one :) : I can set specifics components to a template but if I want thoses components to be load in the template without pushing the "add component button" is that possible? The idea is by example for the homepage to load some component and don't let the opportunities to add other to the user. 
Thanks !

**Reply by comsup** [reply]
I found how, just call  'fields' =>  Components\ListComponents\getACFLayout(), in the fieldsGroups/myComponents.php in replacement of the flexible_content module :)

**Reply by aaronmeder** [reply]
Hmm like a template that is locked? I don't know a proper way to achieve that right away if you still wan to keep the admin interface for editing the Component's content.

If you don't really need the admin interface you could remove the normal pageComponents in wp-admin and show a special ACF Field Group on that page to let you enter some specific content. In the page template then load the values of that field group and pass it to renderComponent (see [example](https://github.com/flyntwp/flynt/blob/master/templates/index.twig)) of these specific components. 

All together feels a bit hacky to me and goes against the idea of having a site built up by dynamic components.
If it's more about adding a clean way to add structured content (like team members with certain information fields) then I would rather create a Custom Post Type with a specific field group in ACF.

**Reply by aaronmeder** [reply]
> I found how, just call 'fields' => Components\ListComponents\getACFLayout(), in the fieldsGroups/myComponents.php in replacement of the flexible_content module :)

Aah you are trying to define a specific set of components to be used, but still in a dynamic/flexible way...then a separate fieldGroup is perfect and very simple and clean 💯

**Reply by comsup** [reply]
Yes the idea is to have flexible component for managing pages by example and let the user choose how he want to show is content but for some specific case like for a product catalog by example, I always want to have a specific group attached to the taxonomy and it seems more logical for me to do it programmatically rather than by creating it from the acf backend and linking it manually. The idea is to automat website duplication by using the component flexibility.. I'm trying to find the best way :)

**Reply by comsup** [reply]
> > I found how, just call 'fields' => Components\ListComponents\getACFLayout(), in the fieldsGroups/myComponents.php in replacement of the flexible_content module :)
> 
> Aah you are trying to define a specific set of components to be used, but still in a dynamic/flexible way...then a separate fieldGroup is perfect and very simple and clean 💯

With this solution I can't load multiple fields from multiple component so I'm thiniking creating a Component for autoloading xx components and calling juste this one for showing as many as Component as I want on some pages.
